src/platform/main.c

```diff
@@ -57,6 +57,18 @@ static void badArgument(const char *arg) {
     printCommandlineHelp();
 }
 
+uint64_t decimalToU64(char *s) {
```

Comment:
> What do you think about using `unsigned long long` as the seed type?

`unsigned long long` guarantees a minimum width of 64 bits but no maximum. For bit-manipulation functions like in crypto and RNG math, I prefer types with an exact size, because it makes clear in the code how many bits are being used (and also modular arithmetic is done in the proper ring).

> Is there a reason we might want to use fixed-width anywhere besides RNG?

I have a feeling that isn't the case, but I'm no C/C++ expert. If a compiler wants to offer larger integers than e.g gcc on x86-64 does, it's unlikely to break anything. The game's behavior would only change if you are already using integer overflow in some places.

Reply:
> Can we not use `sscanf()` with `%llu` or `strtoull()`?

I'd be fine with `sscanf`, although it has some limitations: it stops at the first non-digit character.

```
brogue --seed "123,456,789"
```
starts a game with seed 123. The patch starts the game with seed 123456789.

When parsing a seed that is too big to fit in 64 bits, both `sscanf` and this function start the game without complaining, with a seed that does not look like what was passed.

Shall I change it to `sscanf` for the sake of simplicity?